### PR TITLE
Support translations of nav items and file actions

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
Menu items shall be translated as well.

The proposed solution is far from perfect - but fits the needs.

## How Has This Been Tested?
- switch browser to German
- open the menu
- see the items translated

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: apps/files/src/default.js [hunk 1]
```diff
@@ -10,8 +10,12 @@ import translationsJson from '../l10n/translations.json'
 
 const store = require('./store')
 
+// just a dummy function to trick gettext tools
+function $gettext (msg) {
```
Comment (DeepDiver1975) [inline apps/files/src/default.js]: :nauseated_face:

File: src/components/Menu.vue [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <template>
   <oc-application-menu name="coreMenu" v-model="sidebarIsVisible" @close="sidebarIsVisible = false">
-    <oc-sidebar-nav-item v-for="(n, nid) in nav" :key="nid" :icon="n.iconMaterial" :target="n.route.path">{{ n.name }}</oc-sidebar-nav-item>
+    <oc-sidebar-nav-item v-for="(n, nid) in nav" :key="nid" :icon="n.iconMaterial" :target="n.route.path">{{ $gettext(n.name) }}</oc-sidebar-nav-item>
```
Comment (DeepDiver1975) [inline src/components/Menu.vue]: :nauseated_face:

Comment (LukasHirt) [reply src/components/Menu.vue]: If we need to use it then still in the component, wouldn't it be better with computed property and using $gettext directly there?

Comment (DeepDiver1975) [reply src/components/Menu.vue]: yes - but this is anyhow wrong .....

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4711/

<p>

























</p></details>

File: apps/files/src/default.js [hunk 3]
```diff
@@ -34,37 +38,37 @@ const appInfo = {
...
+    name: $gettext('Shared with others'),
     iconMaterial: 'share',
     route: {
       name: 'files-shared-with-others',
       path: `/${appInfo.id}/shared-with-others`
     }
   },
   {
-    name: 'Deleted files',
+    name: $gettext('Deleted files'),
```
Comment (LukasHirt) [inline apps/files/src/default.js]: Do we need the `$gettext` here when you use it then again in components?

Comment (DeepDiver1975) [reply apps/files/src/default.js]: it is needed here to get the string extracted. the real translation work is done in the component

**Pull request merged**